Jenkinsfile

```diff
@@ -30,17 +30,36 @@ pipeline {
         //         }
         //     }
         // }
-        stage('Go build') {
+        stage('Go build x64') {
             agent {
                 docker {
-                    image docker.build("storj-ci", "--pull https://github.com/storj/ci.git#main").id
-                    args '--user root:root --volume "/tmp/gomod":/go/pkg/mod '
+                    image 'golang:1.17.2'
+                    args '--user root:root --volume /tmp/gomod:/go/pkg/mod '
                 }
             }
             steps {
                 script {
-                    sh './build.sh'
-                    stash(name: "build", includes: "build/")
+                    sh 'make build-x64'
+                    stash(name: "build-x64", includes: "build/")
+                }
+            }
+            post {
+                always {
+                    cleanWs()
+                }
+            }
+        }
+        stage('Go build arm64') {
+            agent {
+                dockerfile {
+                    dir 'docker/go-docker'
+                    args '--user root:root --volume /var/run/docker.sock:/var/run/docker.sock --volume /tmp/gomod:/go/pkg/mod '
```

Comment:
I remember that in the past @stefanbenten was worried about exposing the host machine docker daemon inside of a container.

Reply:
In general the Jenkins setup allows for PR authors to abuse the Jenkins nodes.

Exposing the docker socket adds a vector when the container is compromised but that seems a lot less likely.

Keeping build/test logic out of the Jenkinsfile is very appealing to me. I actually also hope to use docker-in-docker to simplify the stage "PHPUnit" (in the future).

I also made #19 which I hope improves security should a container get compromised.

Reply:
If it's fine for @stefanbenten it's fine for me.